server/app.js

```diff
@@ -118,24 +120,46 @@ app.post("/webhook", (req, res) => {
       payloadToSign,
     });
 
+    // Validates if the timestamp is within an acceptable timeframe
+    // For more information visit https://webhooks.fyi/security/replay-prevention
+    const isTimestampValid = validateTimestamp(revolutRequestTimestamp);
+
+    if (!isTimestampValid) {
+      // The request timestamp is outside of the tolerance zone.
+      return res.status(403).send("Timestamp outside the tolerance zone");
+    }
+
     if (isSignatureValid) {
+      const order = orders.getOrderByRevolutId(req.body["order_id"]);
+
       // Return 200 response before doing any order computation
       res.sendStatus(200);
 
       switch (req.body.event) {
         // Add any order management logic (send emails, update orders, etc)
         case "ORDER_COMPLETED":
           console.log("Webhook - Order Completed!");
+          ordersQueue.push({
+            orderId: order.id,
```

Comment:
is it the same order id as in `req.body["order_id"]`?

Reply:
No, it's not.  This one is the internal one (from merchant's) after the DB call:
`const order = orders.getOrderByRevolutId(req.body["order_id"]);`

But also moved that call inside of the worker.